tests/test_causal_graph.py

```diff
@@ -690,8 +690,47 @@ def test_acyclicity_check_large_adjacency(self):
         # Large dense adjacency matrix
         adj_matrix = numpy.triu(rng.random(size=(400, 400)) > 0.8)
 
-        # Should take ~23s to complete
-        graph = CausalGraph.from_adjacency_matrix(adjacency=adj_matrix)
+        # Should take ~2s to complete
```

Comment:
@steven-causalens to be fair, by moving to single validation at end, this is still much faster. So maybe we just want to default to `validate=True` or remove the parameter are always validate at the end, as opposed to when each edge is added. Food for thought.